Examples/src/main/java/com/mapbox/vision/examples/POIActivity.java

```diff
@@ -0,0 +1,389 @@
+                    result.add(poiState);
+                }
+            }
+            return result;
+        }
+
+        @NonNull
+        private List<POIDrawData> preparePOIDrawData(List<POIState> poiStateList){
+
+            // Prepare bounding rect for POI in mobile screen coordinates
+            final List<POIDrawData> poiDrawDataList = new ArrayList<>();
+            for (POIState poiState: poiStateList) {
+                final Rect poiBitmapRect = calculatePOIScreenRect(poiState.getWorldCoordinate());
+                final int poiLabelAlpha = calculatePOILabelAlpha(poiState);
+                final POIDrawData poiDrawData = new POIDrawData(poiState.getPOI().getBitmap(), poiBitmapRect, poiLabelAlpha);
+                poiDrawDataList.add(poiDrawData);
+            }
+            return poiDrawDataList;
+        }
+
+        private Rect calculatePOIScreenRect(@NonNull WorldCoordinate poiWorldCoordinate) {
+
+            // Calculate left top coordinate of POI in real world using POI world coordinate
+            final WorldCoordinate worldLeftTop = poiWorldCoordinate.copy(
+                    poiWorldCoordinate.getX(),
+                    poiWorldCoordinate.getY() + (float) LABEL_SIZE_METERS / 2,
+                    poiWorldCoordinate.getZ() + LABEL_ABOVE_GROUND_METERS + LABEL_SIZE_METERS
+            );
+
+            // Calculate right bottom coordinate of POI in real world using POI world coordinate
+            final WorldCoordinate worldRightBottom = poiWorldCoordinate.copy(
+                    poiWorldCoordinate.getX(),
+                    poiWorldCoordinate.getY() - (float) LABEL_SIZE_METERS / 2,
+                    poiWorldCoordinate.getZ() + LABEL_ABOVE_GROUND_METERS
+            );
+
+            Rect poiBitmapRect = new Rect(0, 0, 0, 0);
+
+            // Calculate POI left top position on camera frame from real word coordinates
+            PixelCoordinate pixelLeftTop = VisionReplayManager.worldToPixel(worldLeftTop);
+            if (pixelLeftTop == null) {
+                return poiBitmapRect;
+            }
+            poiBitmapRect.left = pixelLeftTop.getX();
+            poiBitmapRect.top = pixelLeftTop.getY();
+
+            // Calculate POI right bottom position on camera frame from real word coordinates
+            PixelCoordinate pixelRightTop = VisionReplayManager.worldToPixel(worldRightBottom);
+            if (pixelRightTop == null) {
+                return poiBitmapRect;
+            }
+            poiBitmapRect.right = pixelRightTop.getX();
+            poiBitmapRect.bottom = pixelRightTop.getY();
+
+            return poiBitmapRect;
+        }
+
+        private int calculatePOILabelAlpha(@NonNull POIState poiState) {
+            int minDistance = Math.min(MIN_DISTANCE_METERS_FOR_DRAW_LABEL - poiState.distanceToVehicle, DISTANCE_FOR_ALPHA_APPEAR_METERS);
+            return (int)((minDistance / (float)DISTANCE_FOR_ALPHA_APPEAR_METERS) * 255);
```

Comment:
should we check `divisionByZero`?

Reply:
I'd avoid. That is an easy to handle even if somebody set initial value to 0 without understanding this variable.

Reply:
okay. let's leave as is